For RHEL4, need to disable coalescing when LRO is enabled,
or get IPoIB performance degradation (TCP stack issue).

Fixes Bugzilla 1494.

Signed-off-by: Jack Morgenstein <jackm@dev.mellanox.co.il>

Index: ofa_kernel-1.4/drivers/infiniband/ulp/ipoib/ipoib_verbs.c
===================================================================
--- ofa_kernel-1.4.orig/drivers/infiniband/ulp/ipoib/ipoib_verbs.c	2009-02-23 11:25:20.000000000 +0200
+++ ofa_kernel-1.4/drivers/infiniband/ulp/ipoib/ipoib_verbs.c	2009-02-23 11:34:39.000000000 +0200
@@ -185,10 +185,17 @@
 
 	coal = kzalloc(sizeof *coal, GFP_KERNEL);
 	if (coal) {
-		coal->rx_coalesce_usecs = 10;
-		coal->tx_coalesce_usecs = 10;
-		coal->rx_max_coalesced_frames = 16;
-		coal->tx_max_coalesced_frames = 16;
+		if (dev->features & NETIF_F_LRO) {
+			coal->rx_coalesce_usecs = 0;
+			coal->tx_coalesce_usecs = 0;
+			coal->rx_max_coalesced_frames = 0;
+			coal->tx_max_coalesced_frames = 0;
+		} else {
+			coal->rx_coalesce_usecs = 10;
+			coal->tx_coalesce_usecs = 10;
+			coal->rx_max_coalesced_frames = 16;
+			coal->tx_max_coalesced_frames = 16;
+		}
 		dev->ethtool_ops->set_coalesce(dev, coal);
 		kfree(coal);
 	}
